contrib/IECoreAlembic/src/IECoreAlembic/MeshWriter.cpp

```diff
@@ -0,0 +1,222 @@
+			{
+				MeshPrimitivePtr meshPrimitiveCopy = typedObject->copy();
+				IECore::MeshAlgo::reverseWinding( meshPrimitiveCopy.get() );
+				meshPrimitive = meshPrimitiveCopy;
+			}
+			else
+			{
+				throw IECore::Exception( "MeshWriter expected a MeshPrimitive" );
+			}
+
+			// If this is the first sample, construct an OPolyMesh or OSubD depending on the mesh
+			// interpolation.
+
+			if( m_state.which() == PendingConstructionState )
+			{
+				const ConstructorArguments &constructorArguments = boost::get<ConstructorArguments>( m_state );
+				if( meshPrimitive->interpolation() == "linear" )
+				{
+					m_state = OPolyMesh( constructorArguments.parent, constructorArguments.name );
+				}
+				else
+				{
+					m_state = OSubD( constructorArguments.parent, constructorArguments.name );
+				}
+			}
+
+			// Now write the sample
+
+			if( m_state.which() == PolyMeshState )
+			{
+				OPolyMeshSchema::Sample sample;
+
+				if( const V3fVectorData *n = meshPrimitive->variableData<V3fVectorData>( "N" ) )
+				{
+					sample.setNormals(
+						ON3fGeomParam::Sample( n->readable(), geometryScope( meshPrimitive->variables.find( "N" )->second.interpolation ) )
+					);
+				}
+
+				writeSampleInternal( meshPrimitive.get(), sample, boost::get<OPolyMesh>( m_state ).getSchema() );
+			}
+			else
+			{
+				OSubDSchema::Sample sample;
+				sample.setSubdivisionScheme( "catmull-clark" );
+				writeSampleInternal( meshPrimitive.get(), sample, boost::get<OSubD>( m_state ).getSchema() );
+			}
+
+		}
+
+		virtual void writeTimeSampling( const Alembic::AbcCoreAbstract::TimeSamplingPtr &timeSampling )
+		{
+			switch( m_state.which() )
+			{
+				case PolyMeshState :
+					boost::get<OPolyMesh>( m_state ).getSchema().setTimeSampling( timeSampling );
+					break;
+				case SubDState :
+					boost::get<OSubD>( m_state ).getSchema().setTimeSampling( timeSampling );
+					break;
+				default :
+					throw IECore::Exception( "MeshWriter::writeSample() must be called before MeshWriter::writeTimeSampling()" );
+			}
+		}
+
+	private :
+
+		template<typename Schema>
+		void writeSampleInternal( const IECore::MeshPrimitive *meshPrimitive, typename Schema::Sample &sample, Schema &schema )
+		{
+
+			sample.setFaceCounts(
+				Abc::Int32ArraySample( meshPrimitive->verticesPerFace()->readable() )
+			);
+
+			sample.setFaceIndices(
+				Abc::Int32ArraySample( meshPrimitive->vertexIds()->readable() )
+			);
+
+			if( const V3fVectorData *p = meshPrimitive->variableData<V3fVectorData>( "P" ) )
+			{
+				sample.setPositions(
+					Abc::P3fArraySample( p->readable() )
+				);
+			}
+
+			const FloatVectorData *sData = meshPrimitive->variableData<FloatVectorData>( "s" );
+			const FloatVectorData *tData = meshPrimitive->variableData<FloatVectorData>( "t" );
+			vector<V2f> st;
+			if( sData && tData )
+			{
+				const vector<float> &s = sData->readable();
+				const vector<float> &t = tData->readable();
+				if( s.size() == t.size() )
```

Comment:
We do. We currently don't load them in MeshReader either, which is why the round-tripping test passes. I didn't see any merit in writing special case code for UVs in both MeshReader and MeshWriter now, when it will happen naturally for all primvars (in PrimitiveReader and PrimitiveWriter) when we do the work to add the indices field to PrimitiveVariable.